tools/publisher/src/subcommand/yank_release.rs

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::cargo;
+use anyhow::{bail, Context, Result};
+use clap::Parser;
+use dialoguer::Confirm;
+use regex::Regex;
+use smithy_rs_tool_common::shell::ShellOperation;
+use smithy_rs_tool_common::versions_manifest::{Release, VersionsManifest};
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+use std::sync::Arc;
+use tokio::sync::Semaphore;
+use tracing::info;
+
+const MAX_CONCURRENCY: usize = 5;
+
+#[derive(Parser, Debug)]
+pub struct YankReleaseArgs {
+    /// The aws-sdk-rust release tag to yank
+    #[clap(long, required_unless_present = "versions-toml")]
+    release_tag: Option<String>,
+    /// Path to a `versions.toml` file with a `[release]` section to yank.
+    /// The `--release-tag` option is preferred to this, but this is provided as a fail safe.
```

Comment:
this is a little confusing—this is a _git_ tag but that's different that the release _version_

I think this can be fixed up with documentation and maybe naming. Something like `--github-tag` maybe? or `--git-tag`? 

As written I briefly thought it would yank all crates that had a specified cargo version